### ISSUE TITLE
Improve Describe-Users Query Capability for WorkDocs

### ISSUE THREAD

**Issue by ccoffman4477** [aws/aws-cli#7000]
### Describe the feature

provide better ability to filter based on WorkDocs user attributes, name query alone is not sufficient

### Use Case

need to describe a specific query to overcome RequestedEntityTooLargeException issues associated with AD Connector and large result sets 

An error occurred (RequestedEntityTooLargeException) when calling the DescribeUsers operation: RequestedEntityTooLargeException: The directory has refused the operation because the result set is too large. Please try a more specific query. Request id: cb800d0f-5705-4be5-9d04-46c27143792e

### Proposed Solution

developed "filters" option to replace "include" and "user-query" which permits matching based on any combination of user attributes (similar to the capability that currently exists in ec2 describe-instances)  

### Other Information

may be unaware the Administrator function in WorkDocs provided to download a user listing also fails silently; no error message and returns zero byte Users.JSON file when directory refuses the operation

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CLI version used

PS >aws --version aws-cli/2.5.4 Python/3.9.11 Windows/10 exe/AMD64 prompt/off

### Environment details (OS name and version, etc.)

PS >aws --version aws-cli/2.5.4 Python/3.9.11 Windows/10 exe/AMD64 prompt/off

**Comment by kdaily** [comment]
Hi @ccoffman4477,

Thanks for writing in! This change would be a feature request for the Workdocs team. The type of filter you're suggesting (similar to `--filters` on the `aws ec2 describe-instances` command) is on the server-side, not the client side. I can pass along your feedback to the Workdocs team, but can't provide any guarantees that they will implement such a feature. 

You noted a separate issue with a silent failure. Can you provide debug logs and more details how you're encountering this behavior? You can add `--debug` to the command you're running. Make sure to redact any sensitive information from the output before posting. Thanks!

**Comment by ccoffman4477** [comment]
hello @kdaily 

Included mention of the silent failure within the WorkDocs Administrator UI to underscore my point no adequate filtering is provided in either the AWSCLI or the Administrator UI function provided for the same purpose.  Administrator's Guide offers instructions within a section titled "Downloading User Lists" under the "Inviting and Managing WorkDocs Users" chapter.

I believe either reducing the group associated with AD Connector to something less than the entire organization or improved filtering capability is the answer to this problem.  Hope the description of the trouble with DescribeUser is clearer with that explanation.

**Comment by tim-finnigan** [comment]
Checking in - as mentioned earlier this feature request would need to be routed to the Workdocs team since service teams own their APIs. I saw this related issue on improving the DescribeUsers documentation: https://github.com/aws/aws-sdk/issues/132. Regarding this request:

> name query alone is not sufficient; develop "filters" option to replace "include" and "user-query" which permits matching based on any combination of user attributes 

I think it's unlikely that the those existing parameters could be replaced because of the need to support backwards compatibility. But if you are requesting additional functionality beyond what is currently available, could you summarize that feature request so that I can forward it to the Workdocs team? As mentioned earlier this request would need to go to them since service teams own their APIs.

**Comment by ccoffman4477** [comment]
hello Tim,

Should filtering be provided, replacement of the existing parameters need not occur until the enhancement is adopted  allowing backward compatibility through many version changes until the old query capability is eventually deprecated.  It's done all the time.  I believe the original solution proposed summarizes the feature request sufficiently.  Asking for the ability to select a subset of the entire AD associated user group based on something other that name.  For example, everyone who has been granted Administrator permissions.

**Comment by tim-finnigan** [comment]
Thanks @ccoffman4477 for following up. I'll link to the CLI documentation on filtering as that is relevant here: https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-filter.html. And I'll reference what was said before:
> This change would be a feature request for the Workdocs team. The type of filter you're suggesting (similar to `--filters` on the `aws ec2 describe-instances` command) is on the server-side, not the client side. I can pass along your feedback to the Workdocs team, but can't provide any guarantees that they will implement such a feature.

I'll transfer this issue to our cross-SDK repository, forward your feedback to the WorkDocs team and post any updates on this issue.
